src/net/virtual_connection.rs

```diff
@@ -61,6 +71,14 @@ impl VirtualConnection {
         self.acknowledge_handler.packets_in_flight() > self.config.max_packets_in_flight
     }
 
+    pub fn is_disconnected(&self) -> bool {
+        if let ConnectionState::Disconnected(_) = self.current_state {
```

Comment:
can't we do something like: `return ConnectionState::Disconnected(_) == self.current_state`

Reply:
I don't know how to do this... because I need to pattern match here. I don't want to compare for equality, because what I care is that state is Disconnected.

Reply:
Okay, don't mind then